tests/evhtests/l7_evh_crd_test.go

```diff
@@ -995,3 +1000,616 @@ func TestHTTPRuleCreateDeleteWithPkiRefForEvh(t *testing.T) {
+	time.Sleep(5 * time.Second)
+
+	_, aviModel = objects.SharedAviGraphLister().Get(modelName)
+	nodes = aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+	g.Expect(nodes[0].EvhNodes[0].SamlSpConfig).To(gomega.BeNil())
+	g.Expect(*nodes[0].EvhNodes[0].SsoPolicyRef).To(gomega.ContainSubstring("thisisaviref-ssopolicyoauth"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.CookieName).To(gomega.Equal("MY_OAUTH_COOKIE"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.CookieTimeout).To(gomega.Equal(int32(120)))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.LogoutURI).To(gomega.Equal("https://auth.com/oauth/logout"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.RedirectURI).To(gomega.Equal("https://auth.com/oauth/redirect"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.PostLogoutRedirectURI).To(gomega.Equal("https://auth.com/oauth/post-logout-redirect"))
+	g.Expect(nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings).To(gomega.HaveLen(1))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].AppSettings.ClientID).To(gomega.Equal("my-client-id"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].AppSettings.ClientSecret).To(gomega.Equal("my-client-secret"))
+	g.Expect(nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].AppSettings.Scopes).To(gomega.HaveLen(1))
+	g.Expect(nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].AppSettings.Scopes[0]).To(gomega.Equal("scope-1"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.OidcEnable).To(gomega.Equal(true))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.Profile).To(gomega.Equal(true))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.Userinfo).To(gomega.Equal(true))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].AuthProfileRef).To(gomega.ContainSubstring("thisisaviref-authprofileoauth"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.AccessType).To(gomega.Equal("ACCESS_TOKEN_TYPE_OPAQUE"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.IntrospectionDataTimeout).To(gomega.Equal(int32(60)))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.OpaqueTokenParams.ServerID).To(gomega.Equal("my-server-id"))
+	g.Expect(*nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.OpaqueTokenParams.ServerSecret).To(gomega.Equal("my-server-secret"))
+	g.Expect(nodes[0].EvhNodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.JwtParams).To(gomega.BeNil())
+
+	// Delete/Disable
+	integrationtest.TeardownSSORule(t, g, sniVSKey, srname)
+	_, aviModel = objects.SharedAviGraphLister().Get(modelName)
+	nodes = aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+
+	g.Expect(nodes[0].EvhNodes[0].SsoPolicyRef).To(gomega.BeNil())
+	g.Expect(nodes[0].EvhNodes[0].OauthVsConfig).To(gomega.BeNil())
+	g.Expect(nodes[0].EvhNodes[0].SamlSpConfig).To(gomega.BeNil())
```

Comment:
Do we have to add `eventually` here since the update won't happen immediately?

Reply:
Don't think it is required here as the fields are getting updated every time. It's the same for hostrule as well.